src/ui/studio/save-creation/index.tsx

```diff
@@ -415,12 +431,49 @@ const UploadForm = ({ uploadState, handleUpload }) => {
     }
   }
 
+  async function handleSelectChange(newValue: SingleValue<SeriesOption>) {
+    let newSeriesValue = newValue ? newValue.value : null;
+    dispatch({ type: 'UPDATE_SERIES', value: newSeriesValue });
+  }
+
+  const filterSeries = async(inputValue: string) => {
+    const seriesList: SeriesOption[] = [];
+    const ocSeries = await opencast.getSeries(inputValue);
+    for (const [key, value] of ocSeries ) {
+      seriesList.push({ value: key, label: value });
+      if (seriesId && seriesId === key) {
+        seriesDefault = { value: key, label: value };
+      }
+    }
+    return seriesList;
+  };
+
+  const lastTimeout = useRef<ReturnType<typeof setTimeout> | null>(null);
+
+  const loadSeriesOptions = (inputValue: string) => {
+    const seriesOptions = new Promise<SeriesOption[]>(resolve => {
+      if (lastTimeout.current !== null) {
+        clearTimeout(lastTimeout.current);
+      }
+      lastTimeout.current = setTimeout(() => {
+        resolve(filterSeries(inputValue));
+      }, 30);
+    });
+    seriesOptions.then(options => {
+      if(seriesId) setSeriesValue(options.find(option => option.value === seriesId));
```

Comment:
```suggestion
      if(seriesId) {
        setSeriesValue(options.find(option => option.value === seriesId));
      }
```

Wow, I really need to add more eslint rules...